test/driver-test/driver-test.cpp

```diff
@@ -0,0 +1,312 @@
+/*
+* driver-test.cpp
+* Nick DeMarco
+* (c) 2020 sigcpp https://sigcpp.github.io. See LICENSE.MD
+*
+* Attribution and copyright notice must be retained.
+* - Attribution may be augmented to include additional authors
+* - Copyright notice cannot be altered
+* Attribution and copyright info may be relocated but they must be conspicuous.
+*
+* Tester for test-stl-lite driver
+*/
+
+#include "../verifiers.h"
+#include "../options.h"
+#include "../tester.h"
+#include "../utils.h"
+
+#include <iostream>
+#include <string_view>
+#include <vector>
+
+void driver_test();
+void test_get_options();
+void reset_default_options(Options& o);
+void compare(const Options& received, const Options& expected, const std::string& test_case);
+std::vector<std::pair<std::string, Options>> generate_cmdline_option_pairs();
+
+void driver_test()
+{
+	test_get_options();
+}
+
+void test_get_options()
+{
+	char* args[100];
+
+	const std::vector<std::pair<std::string, Options>> args_and_options = generate_cmdline_option_pairs();
+	std::size_t cmd_line_number{ 0 };
+	for (auto iter = args_and_options.begin(); iter != args_and_options.end(); ++iter)
+	{
+		std::vector<std::string> v = split(iter->first, ' ');
+		for (std::size_t i = 0; i < v.size(); ++i)
+			args[i] = v[i].data();
+
+		Options o = get_options(args, v.size());
+
+		compare(o, iter->second, "cmd_line_" + std::to_string(cmd_line_number));
+		++cmd_line_number;
+	}
+}
+
+void reset_default_options(Options& o)
+{
+	o.header = true;
+	o.summary = true;
+	o.header_text = "Running $suite";
+	o.prm = pass_report_mode::indicate;
+	o.fail_threshold = 0;
+	o.fom = file_open_mode::no_file;
+	o.output_filepath = "";
+	o.command_name = "array_test";
+	o.suites_to_run = "";
+}
+
+void compare(const Options& received, const Options& expected, const std::string& test_case)
+{
+	is_true(received.header == expected.header, (test_case + " header").data());
+	is_true(received.summary == expected.summary, (test_case + " summary").data());
+	is_true(received.header_text == expected.header_text, (test_case + " header text").data());
+	is_true(received.prm == expected.prm, (test_case + " pass report mode").data());
+	is_true(received.fail_threshold == expected.fail_threshold, (test_case + " fail threshold").data());
+	is_true(received.fom == expected.fom, (test_case + " file open mode").data());
+	is_true(received.output_filepath == expected.output_filepath, (test_case + " output filepath").data());
+	is_true(received.command_name == expected.command_name, (test_case + " command name").data());
+	is_true(received.suites_to_run == expected.suites_to_run, (test_case + " suites to run").data());
+}
+
+std::vector<std::pair<std::string, Options>> generate_cmdline_option_pairs()
```

Comment:
At first look, this function interface seems more complicated than it should be. At the least, this function is easily improved by returning a map instead of a vector of pairs where each pair has a unique string "test case name" and the corresponding options. This is exactly what a map already does with less clutter.

Reply:
@smurthys I thought about using a map, but I chose a vector because a vector gives me the ability to order by when the element was inserted. The dummy cmd-lines are grouped in a logical way that tests related cases together (eg. test single name-values options first, test edge cases last), I wanted the to run the tests in that same logical order. As far as I know, a std::map will sort by the value of the key, and a std::unordered_map does not guarantee any specific order of its elements, thus, iterating through a map to run the tests will not produce the order of tests I was looking for. A map could definitely de-clutter the code, but the look-up and insertion performance that is often the primary pro of using a map does not benefit us here and we would lose the logical order of tests run. Perhaps the pros of using a map outweigh the cons?

Reply:
Good perspective @Ndemco. A few observations: there is no benefit to running the tests in the order written unless there is dependency between tests (which isn't the case here). Even then, `std::map` guarantees order and you can even enforce a specific order.

This said, a vector of pairs or a map is unnecessary as I have outlined in my [other comment](https://github.com/sigcpp/stl-lite/pull/27#issuecomment-645542322). Using the approach in that suggestion is what really reduces clutter, complexity, and complications.